optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/lambda/PropertySetterAccessor.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.domain.common.accessor.lambda;
+
+import java.lang.invoke.CallSite;
+import java.lang.invoke.LambdaConversionException;
+import java.lang.invoke.LambdaMetafactory;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.Method;
+
+public abstract class PropertySetterAccessor<T> {
+
+    private final Method setterMethod;
+    private final Class<?> propertyType;
+    private final MethodHandles.Lookup lookup;
+    private final T setterFunction;
+
+    public static PropertySetterAccessor create(final Method setterMethod,
+                                                final Class<?> propertyType,
+                                                final MethodHandles.Lookup lookup) {
+        if (Object.class.isAssignableFrom(propertyType)) {
+            return new ObjectPropertySetterAccessor(setterMethod, propertyType, lookup);
+        } else if (boolean.class.isAssignableFrom(propertyType)) {
```

Comment:
Are you sure the isAssignableFrom() call is correct/fast? I am thinking unboxing etc (although the Object isAssignable would catch that).

Class has method isPrimitive(), which javadocs mentions Long.TYPE, etc.

In ConfigUtils, we do:

                } else if (propertyType.equals(Boolean.TYPE) || propertyType.equals(Boolean.class)) {
                    typedValue = Boolean.parseBoolean(valueString);
                } else if (propertyType.equals(Integer.TYPE) || propertyType.equals(Integer.class)) {
                    typedValue = Integer.parseInt(valueString);

So, just thinking aloud, would it make more sense to replace:
   `} else if (boolean.class.isAssignableFrom(propertyType)) {`
with
   `} else if (propertyType.equals(Boolean.TYPE)) {`

or vica versa in ConfigUtils? I'd argue the second approach is faster.

Reply:
I didn't consider which approach might be faster; this code is invoked once when the lambda is being created. Thanks for this hint, will replace it.